### PR TITLE
Display plugin in validation messages

### LINKED ISSUE #16904: Report plugin context for validation problems
Currently the messages that we show to a user when a validation problem is detected will give the type name of a task (or transform), but won't tell what plugin contributes this task. This makes it hard for a user to tell if the problem is on their side or not.

---
cc: @gradle/execution

### PR DESCRIPTION
**Pull request by melix**
This reverts commit 4a44f8a155112a1f331a46b06c9e24143c62f61e.

Fixes #16904

### CODE REVIEW

**Comment by melix** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

File: subprojects/execution/src/main/java/org/gradle/internal/execution/impl/DefaultExecutionEngine.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public Request createRequest(UnitOfWork work) {
             private ExecutionRequestContext createExecutionRequestContext() {
                 WorkValidationContext validationContext = this.validationContext != null
                     ? this.validationContext
-                    : new DefaultWorkValidationContext(documentationRegistry);
+                    : new DefaultWorkValidationContext(documentationRegistry, WorkValidationContext.TypeOriginInspector.NO_OP);
```
[the remark this answers is not in the document]
Comment (melix) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/impl/DefaultExecutionEngine.java:50]: So you'd add a default method on `UnitOfWork` to get the type origin inspector returning `NO_OP`?

[the remark this answers is not in the document]
Comment (lptr) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/impl/DefaultExecutionEngine.java:50]: That would be a consequence, yes.

**Comment by melix** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

**Review by lptr — APPROVED** [review]
LGTM.

**Comment by melix** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
I have fixed the verification problem on master.

@bot-gradle test and merge

**Comment by melix** [comment]
@bot-gradle test and merge

**Comment by melix** [comment]
@bot-gradle test and merge

**Pull request merged**